test/test_autograd.py

```diff
@@ -1221,6 +1221,34 @@ def test_backward_no_grad(self):
         with self.assertRaises(RuntimeError):
             torch.autograd.backward([b], [None])
 
+    def test_backward_twice_with_saved_values(self):
+        b = torch.randn(3, requires_grad=True, dtype=torch.double)
+        c = torch.zeros(3, dtype=torch.double)
+        c[[1, 2]] = b[[1, 1]]
```

Comment:
It's not completely obvious that this indexing expression causes a `TensorList` to be saved for backwards and that is what we're fixing in this PR (that backward twice with a saved TensorList works as expected). I don't see any other functions in derivatives.yaml that saves a TensorList, though, so I don't have ideas on how to make this test better.